src/main/java/dev/dbos/transact/workflow/internal/WorkflowHandleFuture.java

```diff
@@ -27,7 +27,7 @@ public String getWorkflowId() {
   }
 
   @Override
-  public T getResult() {
+  public T getResult() throws E {
```

Comment:
should we be matching the code in `WorkflowHandleDBPoll` and cast the thrown exception to `E` rather than throwing an unexpected `DBOSException`?